lnclient/ldk/ldk.go

```diff
@@ -1588,6 +1587,43 @@ func (ls *LDKService) saveStaticChannelBackupToDisk(event *events.StaticChannels
 	logger.Logger.WithField("backupPath", backupFilePath).Debug("Saved static channel backup to disk")
 }
 
+func (ls *LDKService) cleanupStaticChannelBackupFiles() {
+	const keepBackupFiles = 3
+
+	backupDirectory := filepath.Join(ls.workdir, "static_channel_backups")
+	logger.Logger.WithField("backupDirectory", backupDirectory).Debug("Deleting old static channel backup files")
+	files, err := os.ReadDir(backupDirectory)
+	if err != nil {
+		logger.Logger.WithField("path", backupDirectory).WithError(err).Error("Failed to list static channel backups directory")
+		return
+	}
+
+	backupFiles := make([]fs.DirEntry, 0, len(files))
+
+	for _, file := range files {
+		name := file.Name()
+		if strings.HasPrefix(name, "20") && filepath.Ext(name) == ".json" && !file.IsDir() {
+			backupFiles = append(backupFiles, file)
+		}
+	}
+
+	if len(backupFiles) <= keepBackupFiles {
+		return
+	}
+
+	// ReadDir() returns file entries listed alphabetically, so we can just
```

Comment:
This is a bit scary to me, can we explicitly sort it in an external function that we can add unit tests for to ensure we only delete the oldest files?

Reply:
@rdmitr I think actually Bumi is right, maybe we convert this to draft for now and just stop saving new backups on every app start as a first step. I will create a separate PR for this.